### ISSUE TITLE
Repeated crashing

### ISSUE THREAD

**Issue by OverLordGoldDragon** [spyder-ide/spyder#20193]
It's mostly fine but occasionally it'll randomly freeze and require restarting. I've attached a bunch of logs from `--debug-info verbose` near the end of it happening, hope it helps. Environment info in [this post](https://github.com/spyder-ide/spyder/issues/20192).

Today for the first time it kept freezing as soon as it loaded up, and kept doing that even after restarting Anaconda Powershell Prompt. I only fixed it via `--new-instance`. Really unsure what's causing it, but possibly Editor activity or `[spyder.plugins.projects.utils.watcher]` when files are frequently modified or created.

[log0.txt](https://github.com/spyder-ide/spyder/files/10217383/log0.txt)
[log1.txt](https://github.com/spyder-ide/spyder/files/10217387/log1.txt)
[log2.txt](https://github.com/spyder-ide/spyder/files/10217386/log2.txt)
[log3.txt](https://github.com/spyder-ide/spyder/files/10217384/log3.txt) -- freeze 'immediately', full log from start to finish

**Comment by ccordoba12** [comment]
Hey @OverLordGoldDragon, thanks for reporting. I also noticed this problem and have ready a small improvement for the bug that appears on `log1.txt` and `log2.txt`, which will be part of 5.4.1.

The problem I can see on `log3.txt` is that you're copying a lot of `npy` files to a folder inside a Spyder project. That requires a bit more of work to solve, so I'll try to address it in 5.4.2.

**Comment by OverLordGoldDragon** [comment]
@ccordoba12 Thanks, had no idea there's such interaction with the project directory but makes sense. If there's other such caveats, might be worth adding to [docs](https://docs.spyder-ide.org/current/panes/projects.html).

**Comment by ccordoba12** [comment]
> Thanks, had no idea there's such interaction with the project directory but makes sense

Yep. I forgot to mention that the problem that appears in `log1.txt` and `log2.txt` is also caused by using a project.

> If there's other such caveats, might be worth adding to [docs](https://docs.spyder-ide.org/current/panes/projects.html).

Good idea, but these are actually bugs that we need to solve so that our projects functionality works as expected. If we don't manage to do it, then we'll add the corresponding warnings.

**Comment by OverLordGoldDragon** [comment]
Quick post to inform that clicking on the `.exec` in Profiler below has frozen Spyder twice in a row, once from fresh start. Maybe it's a long list and it struggles.

![image](https://user-images.githubusercontent.com/16495490/211159751-6d1a6494-43d9-4f66-92a5-8de5caac9366.png)

**Comment by OverLordGoldDragon** [comment]
Started Spyder, froze indefinitely after few minutes of doing nothing, _without_ a project - logs cut off (too much text):

[log4.txt](https://github.com/spyder-ide/spyder/files/10472005/log4.txt)

That's 5.4.1, here's right after upgrading to 5.4.2, no cutoff:

[log5.txt](https://github.com/spyder-ide/spyder/files/10472017/log5.txt)

**Comment by OverLordGoldDragon** [comment]
5.3.0 is the latest version that stopped freezing, at least at startup as far as I can tell now. I tried every version from 5.4.2.

**Comment by kevin-guimard-ext** [comment]
I also experience random and frequent freezes in Spyder's console since a few weeks. Though, I don't know whether it's the same issue reported by @OverLordGoldDragon since it does not require restarting. Spyder hangs for something like 20s-30s before it gives back control. I've been unable to identify what causes these freezes, it even happens when I call "clear" on the console.

**Comment by OverLordGoldDragon** [comment]
I'll just note that it might have been a configurations issue (`spyder.ini`, `transient.ini`) - I've reproduced it once by switching to old configurations on a new computer, Windows 11, but couldn't reproduce it again. I've had no issues with different configs since then. I still have the old configs, if needed.

**Comment by kevin-guimard-ext** [comment]
Well, the only things I changed in the configuration were: the shortcut to run code in a selection and the working directory. And yet, it was by using the UI menus, I didn't change the ini files manually.

**Comment by ccordoba12** [comment]
Hey @kevin-guimard-ext, sorry to hear that. I already have several fixes to address this problem, which will be part of Spyder 6, to be released before the end of the year.

Sorry for the inconvenience in the meantime.

**Comment by kevin-guimard-ext** [comment]
Thank you @ccordoba12.

**Comment by mluerig** [comment]
I have the same issue in 6.0.1 - freezes every few seconds, script size doesn't matter. only in projects though...

**Comment by ccordoba12** [comment]
@mluerig, sorry to hear that. Please follow the instructions I posted in https://github.com/spyder-ide/spyder/issues/22556#issuecomment-2374716821 to disable first our `fzf` support and, if that doesn't work, our file system watcher, to see if that fixes your problem.

Let us know which one works for you.

**Comment by mluerig** [comment]
> @mluerig, sorry to hear that. Please follow the instructions I posted in https://github.com/spyder-ide/spyder/issues/22556#issuecomment-2374716821 to disable first our fzf support and, if that doesn't work, our file system watcher, to see if that fixes your problem.

this worked for me, and above all, in v6.0.2 the issue appears to be gone - haven't noticed it at least

**Comment by mluerig** [comment]
sorry for the late reply. the first fix works (`self._fzf = None`), but it doesn't seem to be fixed in 6.0.2?
